node/src/BaseClient.ts

```diff
@@ -4136,6 +4138,59 @@ export class BaseClient {
         return this.createWritePromise(createZIncrBy(key, increment, member));
     }
 
+    /**
+     * Iterates incrementally over a sorted set.
+     *
+     * See https://valkey.io/commands/zscan for more details.
+     *
+     * @param key - The key of the sorted set.
+     * @param cursor - The cursor that points to the next iteration of results. A value of "0" indicates the start of
+     *      the search.
+     * @param options - (Optional) The zscan options.
+     * @returns An `Array` of the `cursor` and the subset of the sorted set held by `key`.
+     *      The first element is always the `cursor` for the next iteration of results. `0` will be the `cursor`
+     *      returned on the last iteration of the sorted set. The second element is always an `Array` of the subset
+     *      of the sorted set held in `key`. The `Array` in the second element is always a flattened series of
+     *      `String` pairs, where the value is at even indices and the score is at odd indices.
+     *
+     * @example
+     * ```typescript
+     * // Assume "key" contains a sorted set with multiple members
+     * let resultCursor = "1";
+     * let newCursor = "0";
+     * let result = [];
+     *
+     * while (resultCursor !== "0") {
+     *      result = await client.zscan("key", newCursor, {
+     *          match: "*",
+     *          count: 5,
+     *      });
+     *      newCursor = result[0];
+     *      console.log("Cursor: ", newCursor);
+     *      console.log("Members: ", result[1]);
+     *
+     *      if (newCursor === "0") {
+     *          break;
+     *      }
+     *
+     *      resultCursor = newCursor;
+     * }
+     * // Cursor:  123
+     * // Members:  ['value 163', '163', 'value 114', '114', 'value 25', '25', 'value 82', '82', 'value 64', '64']
+     * // Cursor:  47
+     * // Members:  ['value 39', '39', 'value 127', '127', 'value 43', '43', 'value 139', '139', 'value 211', '211']
+     * // Cursor:  0
+     * // Members:  ['value 55', '55', 'value 24', '24', 'value 90', '90', 'value 113', '113']
+     * ```
+     */
+    public async zscan(
+        key: string,
+        cursor: string,
+        options?: BaseScanOptions,
```

Comment:
```suggestion
        options?: ZScanOptions,
```

Reply:
Based on our discussion. We are keeping it as is and add anonymous types for other scan commands. So it would be something similar to `options?: BaseScanOptions & { otherOption: string }`.